lsp-ui-imenu.el

```diff
@@ -331,7 +354,40 @@ Return the updated COLOR-INDEX."
 (define-derived-mode lsp-ui-imenu-mode special-mode "lsp-ui-imenu"
   "Mode showing imenu entries.")
 
-(defun lsp-ui-imenu-enable (_enable))
+(defun lsp-ui-imenu--refresh ()
+  "Safe refresh imenu content."
+  (interactive)
+  (let ((imenu-buffer (get-buffer lsp-ui-imenu-buffer-name)))
+    (when imenu-buffer
+      (save-selected-window
+        (if (equal (current-buffer) imenu-buffer)
+            (select-window (get-buffer-window lsp-ui-imenu--origin))
+          (setq lsp-ui-imenu--origin (current-buffer)))
+        ;; Force refresh, ignore custom variable.
+        (let ((lsp-ui-imenu-auto-refresh t)) (lsp-ui-imenu))))))
```

Comment:
This shouldn't call `lsp-ui-imenu`, `lsp-ui-imenu` does extra work to create a window.
You will need to split `lsp-ui-imenu`

Reply:
Okay, I have extracted the refresh execution to `lsp-ui-imenu--refresh-content` function.